feat: 1448 apply user animation preference on otter generated application (#266)
AmadeusITGroup/otter: docs/application/REDUCED_MOTION.md

# Reduced Motion Setup
The ``@o3r/application`` module provides ``prefersReducedMotion()`` -- a helper that identifies the computer animation preferences.

> This setup is generally used in the scope of accessibility as some users find difficult to interact with pages with many 

Reviewer: ```suggestion
This setup is generally used in the scope of accessibility as some users find it difficult to interact with pages with many 
```

animations.

> To respect their choice, you can manage your application angular animation activation to follow the computer preferences.

Reviewer: lots of "-tion" in a row, what do you think of this suggestion ? 
```suggestion
To respect their choice, you can manage the activation of your application's angular animation to follow the computer preferences.
```

> You just need modify the ``BrowserAnimationsModule`` import in your application module file as follows:

Reviewer: ```suggestion
You just need to modify the ``BrowserAnimationsModule`` import in your application module file as follows:
```

```typescript
@NgModule({
  imports: [
    //...
    BrowserAnimationsModule.withConfig({
      disableAnimations: prefersReducedMotion()
    }),
    //...
    ]
})
export class AppModule {}
```

> **Note:** Any new application that install the ``@o3r/core`` via the ``ng add @o3r/core`` command will follow this recommendation.

Reviewer: ```suggestion
**Note:** Any new application that installs the ``@o3r/core`` via the ``ng add @o3r/core`` command will follow this recommendation.
```

> **Note:** If you face timing issue in flaky e2e tests that might be linked to animation delays, you can force the reduced

Reviewer: ```suggestion
**Note:** If you face timing issues in flaky e2e tests that might be linked to animation delays, you can force the reduced
```

# Reduced Motion Setup
The ``@o3r/application`` module provides ``prefersReducedMotion()`` -- a helper that identifies the computer animation preferences.
This setup is generally used in the scope of accessibility as some users find it difficult to interact with pages with many animations.

To respect their choice, the application's angular animations should follow the computer preferences.

This can be done in the BrowserAnimationsModule's configuration as follows:
```typescript
// Other imports
import {prefersReducedMotion} from '@o3r/application';

@NgModule({
  imports: [
    //...
    BrowserAnimationsModule.withConfig({
      disableAnimations: prefersReducedMotion()
    }),
    //...
    ]
})
export class AppModule {}
```

> **Note:** Any new application that installs the ``@o3r/core`` via the ``ng add @o3r/core`` command will follow this recommendation.

Reviewer 2: It is not part of a `ng update` on purpose?

Author: As discussed in the original story, the aim is to apply our recommandation on the new applications. I could add an ng-update but is it something we want to apply on new project without forewarning?
